contracts/proxies/CurrencyConvertor.sol

```diff
@@ -77,37 +77,48 @@ contract CurrencyConvertor is BaseRelayRecipient {
   }
 
   /**
-    * @notice Make a deposit to the Starkware Layer2 Solution
+    * @notice Approve the token to swap and then makes a deposit with said token.
+    * @dev Emits LogConvertedDeposit event.
     *
-    * @param  depositAmount      The amount of USDC to deposit.
+    * @param  tokenFrom          The token to convert from.
+    * @param  tokenFromAmount    The amount of `tokenFrom` tokens to deposit.
+    * @param  minUsdcAmount      The minimum USDC amount the user will accept in a swap.
     * @param  starkKey           The starkKey of the L2 account to deposit into.
     * @param  positionId         The positionId of the L2 account to deposit into.
+    * @param  exchange           The exchange being used to swap the taker token for USDC.
+    * @param  allowanceTarget    The address being approved for the swap.
+    * @param  data               Trade parameters for the exchange.
     */
-  function deposit(
-    uint256 depositAmount,
+  function approveSwapAndDepositERC20(
```

Comment:
Why is there no function combining register + `approveSwapAndDepositERC20`?

Related: since this is getting to be a lot of functions, instead of having separate functions for register, I would just add the registration signature param, and register the user if the param has nonzero length